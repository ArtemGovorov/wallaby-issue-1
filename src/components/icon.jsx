import React from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const Icon = ({ faIcon }) => <FontAwesomeIcon icon={faIcon} />

Icon.propTypes = {
  faIcon: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
}
