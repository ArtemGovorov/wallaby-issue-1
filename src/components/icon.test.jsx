import React from 'react'
import { cleanup, render } from 'react-testing-library'

import { Icon } from './icon'

describe('icon', () => {
  afterEach(cleanup)
  const defaultProps = {
    faIcon: 'faIcon',
  }

  test('should render', () => {
    const { container } = render(<Icon {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
