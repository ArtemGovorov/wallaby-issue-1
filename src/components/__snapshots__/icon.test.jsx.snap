// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon should render 1`] = `
<i
  id="MockFontAwesomeIcon"
/>
`;
